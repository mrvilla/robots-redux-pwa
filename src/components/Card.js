import React from 'react';

const Card = ({id, name, email}) => {
  return (
    <div className="dib br3 pa2 ma2 tc">
      <img src={`https://robohash.org/${id}?size=220x220`} alt='robot' />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
