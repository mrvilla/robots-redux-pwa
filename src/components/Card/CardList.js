import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  // if (true) {
    // throw new Error('Oh Shit! Sorry bruh');
  // }

  return (
    <div className='tc'>
      {robots.map((user, idx) => {
        return <Card
          key={idx}
          id={robots[idx].id}
          name={robots[idx].name}
          email={robots[idx].email}
        />
      })}
    </div>
  );
};

export default CardList;
