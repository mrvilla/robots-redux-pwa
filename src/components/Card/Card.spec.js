import React from "react";
import renderer from "react-test-renderer";
import Card from './Card';

describe("Card Snapshot testing", () => {
  it("Card renders correctly", () => {
    const domTree = renderer.create(<Card />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
