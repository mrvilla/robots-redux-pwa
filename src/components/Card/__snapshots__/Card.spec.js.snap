// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot testing Card renders correctly 1`] = `
<div
  className="dib br3 pa2 ma2 tc"
>
  <img
    alt="robot"
    src="https://robohash.org/undefined?size=220x220"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
